client/lib/abtest/index.js

```diff
@@ -133,28 +154,34 @@ ABTest.prototype.isEligibleForAbTest = function() {
 	const clientLanguage = client.language || client.userLanguage || 'en';
 	const clientLanguagesPrimary = ( client.languages && client.languages.length ) ? client.languages[ 0 ] : 'en';
 	const localeFromSession = i18n.getLocaleSlug() || 'en';
-	const englishMatcher = /^en-?/i;
 
 	if ( ! store.enabled ) {
 		debug( '%s: Local storage is not enabled', this.experimentId );
 		return false;
 	}
 
-	if ( ! this.allowAnyLocale ) {
-		if ( isUserSignedIn() && user.get().localeSlug !== 'en' ) {
-			debug( '%s: User has a non-English locale', this.experimentId );
-			return false;
+	if ( this.localeTarget ) {
+		const localeMatcher = new RegExp( '^' + this.localeTarget + '-?', 'i' );
+		let isTargetLocale = true;
+
+		if ( isUserSignedIn() && user.get().localeSlug !== this.localeTarget ) {
```

Comment:
I realize the prior code didn't do this either, but why not make use of `localeMatcher` for a more general `en-` match here (e.g., capture `en-gb` and `en-ca`, etc.)?